components/admin/lessons/AdminLessonSideNav.tsx

```diff
@@ -0,0 +1,129 @@
+
+type Items = Item[]
+
+type Props = {
+  items: Items | undefined
+  title: string
+  lessonId: number
+  onAddItem: (m: Item | undefined) => void
+  loading?: boolean
+  error?: ApolloError
+  active?: number
+  onClick: (item: Item) => void
+}
+
+const AdminLessonSideNav = ({
+  items,
+  onAddItem,
+  loading,
+  error,
+  active,
+  title,
+  lessonId,
+  onClick
+}: Props) => {
+  const itemsList =
+    items &&
+    items.reduce((acc: JSX.Element[], item) => {
+      if (item.lesson.id === lessonId) {
+        const isActive = active === item.id
+        const className =
+          styles[
+            isActive
+              ? 'container__modulesList__module--active'
+              : 'container__modulesList__module'
+          ]
+        const withChevron = !isActive && <ChevronRightIcon size={16} />
+
+        acc.push(
+          <li className={className} key={item.id} onClick={() => onClick(item)}>
+            <span>{item.name}</span>
+            {withChevron}
+          </li>
+        )
+      }
+
+      return acc
+    }, [])
+
+  const renderItems = () => {
+    if (error) {
+      return (
+        <div className={styles.container__error}>
+          <AlertFillIcon />
+          <span className={styles.container__heading}>
+            Failed to get the modules
+          </span>
+        </div>
+      )
+    }
+
+    if (loading || lessonId < 0) {
+      return (
+        <>
+          <span className={styles.container__heading}>Existing {title}</span>
+          <ol
+            className={`${styles.container__modulesList} ${styles['container__modulesList--loading']}`}
+          >
+            <Spinner animation={'border'} role="status">
+              <span className="visually-hidden">Loading...</span>
+            </Spinner>
+          </ol>
+          <button className={styles['container--button']}>
+            ADD NEW MODULE
+          </button>
+        </>
+      )
+    }
+
+    const onClick = () => isNumber(active) && onAddItem(get(items, active))
+
+    if (get(itemsList, 'length')) {
+      return (
+        <>
+          <span className={styles.container__heading}>Existing {title}</span>
+          <ol className={styles.container__modulesList}>{itemsList}</ol>
+          <button onClick={onClick} className={styles['container--button']}>
+            ADD NEW MODULE
+          </button>
+        </>
+      )
+    }
+
+    return (
+      <>
+        <span className={styles.container__heading}>Existing {title}</span>
+        <ol className={styles.container__modulesList}>
+          <div className={styles.container__error}>
+            <AlertFillIcon />
+            <span className={styles.container__heading}>
+              No modules in this lesson
+            </span>
+          </div>
+        </ol>
+
+        <button onClick={onClick} className={styles['container--button']}>
+          ADD NEW MODULE
+        </button>
+      </>
+    )
+  }
+
+  return <div className={styles.container}>{renderItems()}</div>
```

Comment:
a few things going on here that feels wrong.

1. I would avoid running a function that returns a component inside a component, it is a bit of an antipattern because a component is a function that returns a component, you could have broken this into a separate component like `<Content ... />` or something. But I wouldn't recommend it for this case because of the next point
2. You created a `renderItems` function to save repeating 1 line of code `<div className={styles.container}>`, which does not help much. It is cleaner to just do:

```
if (condition1) {
  return (
    <div className={styles.container}>
       ...
    </div>
  )
}
if (condition2) {
  return (
    ...
  )
}
...
```